src/QMCWaveFunctions/Fermion/DiracDeterminantBatched.h

```diff
@@ -55,23 +63,25 @@ template<typename DET_ENGINE = MatrixUpdateOMPTarget<QMCTraits::ValueType, QMCTr
 class DiracDeterminantBatched : public DiracDeterminantBase
 {
 public:
-  using ValueVector_t = SPOSet::ValueVector_t;
-  using ValueMatrix_t = SPOSet::ValueMatrix_t;
-  using GradVector_t  = SPOSet::GradVector_t;
-  using GradMatrix_t  = SPOSet::GradMatrix_t;
-  using HessMatrix_t  = SPOSet::HessMatrix_t;
-  using HessVector_t  = SPOSet::HessVector_t;
-  using HessType      = SPOSet::HessType;
-
-  using mValueType = QMCTraits::QTFull::ValueType;
-  using mGradType  = TinyVector<mValueType, DIM>;
-
+  using WFT      = typename DET_ENGINE::WFT;
```

Comment:
And the work toward having a single executable for full and mixed precision real and complex as well as testing all those every time is preserved.

Reply:
If WFC, DiracDeterminantBase, SPOSet still have configuration time locked types, I doubt all 4 listed variants of DiracDeterminantBatched be compiled in one short. Due to existing constraints, you cannot prove this scheme works for the whole code and I cannot defend this scheme doesn't work for the whole code.
So I prefer a separate study on small scale code to look at all the propagation.

Reply:
I don't want to leave you struggling with types like last attempt. We need to look at the issue more thoroughly on an optimal path with strong determination.

Reply:
My last attempt I was blocked, it wasn't a struggle no progress toward one executable or testing complex and real in the same build was made. Possibly correctly templating from the Drivers down on precision and complex was declared although I proved several times in miniqmc it worked fine. This is from the other direction.

Yes SPOSet ruins this because even though it could be _very_ easily templated on Value but it isn't. It has its own type mess, OrbitalTraits which defines a a huge number of types it doesn't have to. I think my way is better and once proven in the code I think we'll want to delete OrbitalTraits and have a templated SPOSet that composes WFT as needed. But SPOSet is out of scope for this PR.

I was only getting clashes with the LogValues, CurRatio, and PsiValue.  I note all WaveFunctionTypes appear to be derivable from the Value and FullPrecValue, I don't know and am not sure yet that the same is true for QMCHamiltonian or elsewhere. 

I rationalize this to the composable WaveFunctionTypes "type class." Since it is the computation layer that must be specialized on precision and real/complex it makes sense that it especially have those parameters.  DiracDeterminantBatched really shouldn't care and shouldn't have data members so it can get those type definitions for its composed determinant engine until we refactor those members to where they belong.

Why do I want to do this because I have to look at them again and again and again and deal with them breaking when different combinations of ENABLE_CUDA and ENABLE_OFFLOAD and QMC_MIXED are used. 

If someone else has a better Idea do it in a part of the code you're working on and if its better it can replace this, I'm tired of abandoning code because I first must prove it will work for everything with no further changes. It's definitely better than the current method. It doesn't break other code. I'd like to get my other work in it's easier and cleaner with this.

I do hope I can propagate composed type aliases based on the minimal number of template parameters through the code over a series of PR's. But I just need them here. I can look in one file and see, when I am working on determinants with Value=X and FullPrecValue=Y what the other types will be I don't need to inherit and I don't need a large number of template parameters.

I did prove many things could work in miniqmc and that work was ignored, I'm not going to go and do that again.  Recall I had real and complex, Kokkos, CUDA, and openmp working in one executable for a couple of weeks before divergence in tracking @ye-luo  and @lshulen branches broke it so many times I gave up.

Reply:
I feel the key issue you are facing is the unclear precision of "LogValues, CurRatio, and PsiValue" propagation. You could have asked for clarification. Trying to workaround existing things without enough understanding is never easy and potentially harmful. You may just make ENABLE_CUDA and ENABLE_OFFLOAD with full precision working and post a PR and ask for help.